### PR TITLE
Few improvements

### PR DESCRIPTION
**Pull request by nirmal070125**
- adds a utility method
- remove name field from the FHIRTypedSearchParameter as it duplicates the search param name

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: base/Dependencies.toml [hunk 1]
```diff
@@ -278,7 +278,7 @@ modules = [
 [[package]]
 org = "ballerina"
 name = "task"
-version = "2.3.3"
+version = "2.3.2"
```
Comment (nirmal070125) [inline base/Dependencies.toml:281]: This is reverted in another commit but it's not getting reflected due to a github downtime. Ignore for now.

File: base/utils.bal [hunk 2]
```diff
@@ -174,3 +175,17 @@ public isolated function createFhirBundle(BundleType t, DomainResource[] resourc
     bundle.entry = entry;
     return bundle;
 }
+
+public isolated function getUrlEncodedSearchParameters(map<RequestSearchParameter[] & readonly> & readonly params)
+returns string|FHIRError {
```
Comment (sameeragunarathne) [inline base/utils.bal:180]: formatting issue?

File: base/utils.bal [hunk 3]
```diff
@@ -174,3 +175,17 @@ public isolated function createFhirBundle(BundleType t, DomainResource[] resourc
...
+    string|error encoded = url:encode(params.toJsonString(), "UTF-8");
+    if (encoded is error) {
+        string errorMsg = "Error occurred while encoding the search parameters.";
+        log:printError(errorMsg, 'error = encoded);
+        FHIRError fhirError = createFHIRError(errorMsg, CODE_SEVERITY_FATAL, INVALID,
```
Comment (sameeragunarathne) [inline base/utils.bal:185]: Let's return inline

File: base/Dependencies.toml (hunk 1 above)
Comment (nirmal070125) [reply base/Dependencies.toml:281]: Ok, seems these versions are not available in Central. I'm committing what I get by building using Central.

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**